src/Resources/npm-package/js/contao-filter-bundle.js

```diff
@@ -52,11 +52,28 @@ class FilterBundle {
             function(element, event) {
                 event.preventDefault();
 
-                let buttonName = element.form.name + '[submit]',
-                    clickedButton = document.createElement('div');
-                clickedButton.setAttribute('name', buttonName);
+                FilterBundle.initAsyncFormSubmit(element);
+            });
+
+        EventUtil.addDynamicEventListener('input',
+            '.mod_filter form[data-async] input[data-submit-on-input], .mod_filter form[data-async] [data-submit-on-input] input',
+            function(element, event) {
+                event.preventDefault();
 
-                FilterBundle.asyncSubmit(element.form, clickedButton);
+                if(element.dataset.threshold > element.value.length) {
+                    return;
+                }
+
+                setTimeout(function() {
+                    element.form.classList.add('keep-form');
+                    FilterBundle.initAsyncFormSubmit(element);
+                }, element.dataset.debounce);
```

Comment:
Wir müssten eventuell schon vorher prüfen, ob wir noch im Debouce-Zeitraum sein, sonst sammeln sich die Eingaben und werden trotzdem nacheinander nach der Zeit abgearbeitet. Zumindest würde ich das hier annehmen?

Reply:
So wie ich das sehe, werden alle Eingaben abgearbeitet unabhängig davon ob in der Debounce-Zeit ein neuer Request abgesendet wurde oder nicht. Ein State das anzeigt ob ein Nachfolgender Request abgesendet wurde, und die Ausführung der vorherigen request verhindert?

Reply:
Ok, ich habs jetzt so umgebaut, dass immer nur der letzte input für den request genutzt wird. Also für den Fall dass in einem debounce ein neuer Request kommt werden die vorhergehenden ignoriert. Nach ein bisschen testen bin ich auch der Meinung, dass das default und nicht konfigurierbar sein muss.

Reply:
Das ist auf jeden Fall schon besser. Ich denke auch, dass diese Verhalten nicht konfigurierbar sein muss. @AlexejKossmann  Ist die Implementation aus deiner Sicht so ok? Bin mir nicht sicher, ob man auch noch gegenprüfen müsste, ob bereit ein AJAX-Request läuft.

Reply:
Ja von meiner Seite sieht es auch gut aus. Ob ein AJAX-Request bereits läuft ist ja egal, ein neuer Request kommt erst nach debounce, aus meiner Sicht ist das richtig so

Reply:
Ok, dann bin ich zufrieden :D